packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -181,6 +181,18 @@ async function instantiateModule(
     }
   }
 
+  let sourceMapSuffix = ''
+  if (result.map) {
+    const moduleSourceMap = Object.assign({}, result.map, {
+      // offset the first three lines of the module (function declaration and 'use strict')
+      mappings: ';;;' + result.map.mappings,
+    })
+    // The ${'//'} is to avoid being replaced by node/server/transformRequest
+    sourceMapSuffix = `\n${'//'}# sourceMappingURL=data:application/json;charset=utf-8;base64,${Buffer.from(
+      JSON.stringify(moduleSourceMap),
+    ).toString('base64')}`
```

Comment:
Maybe we should skip `ssrFixStacktrace` when sourcemaps are enabled? I guess `ssrFixStacktrace` will break the stacktrace by rewriting the stacktrace using sourcemaps again. But I failed to reproduce this. Perhaps I'm missing something.

Reply:
I'm not sure what you mean, can you say more?

Reply:
When Vite is run without `--enable-source-maps`, the stacktrace points to the generated code's positions.

This `ssrFixStacktrace` maps the generated code's positions to the original code's positions by sourcemap.
https://github.com/vitejs/vite/blob/a56bc3434e9d4bc7f9d580ae630ccc633e7d436a/packages/vite/src/node/ssr/ssrModuleLoader.ts#L206-L207
(This part only `fixStacktrace: true` is passed to `viteServer.ssrLoadModule`. Can be run independently by `viteServer.ssrFixStacktrace`)

But after this PR and Vite is run with `--enable-source-maps`, Node.js will generate the stacktrace that is already mapped by sourcemap. So running `ssrFixStacktrace` on that will cause double re-mapping IIUC.